src/shared_memory.rs

```diff
@@ -0,0 +1,12 @@
+pub trait ShareableMemory {
+    fn driver_number(&self) -> usize;
+
+    fn allow_number(&self) -> usize;
+
+    fn to_bytes(&self) -> &[u8];
```

Comment:
Now, that I start thinking about it: The signature should be `(&mut self) -> &mut [u8]`